module/module.go

```diff
@@ -31,17 +32,50 @@ import (
 	rutils "go.viam.com/rdk/utils"
 )
 
-// CheckSocketAddressLength returns an error if the socket path is too long for the OS.
-func CheckSocketAddressLength(addr string) error {
+const (
+	socketSuffix = ".sock"
+	// If we assume each robot has 10 modules running on it, and they would all have colliding truncated names,
+	// P(collision) = 1-(58^5)!/((58^5-10)!*(58^(5*10))) ~ 6.8E-8.
+	socketRandomSuffixLen int = 5
+	base58Chars               = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"
+)
+
+// TruncatedSocketAddress returns a socket address of the form parentDir/desiredName.sock
+// if it is shorter than the max socket length on the given os. If this path would be too long, this function
+// truncates desiredName and returns parentDir/truncatedName-randomStr.sock.
+func TruncatedSocketAddress(parentDir, desiredName, goos string) (string, error) {
 	// maxSocketAddressLength is the length (-1 for null terminator) of the .sun_path field as used in kernel bind()/connect() syscalls.
-	maxSocketAddressLength := 103
-	if runtime.GOOS == "linux" {
-		maxSocketAddressLength = 107
+	maxSocketAddressLength := func() int {
```

Comment:
New thought. How about just skip this entirely? Just set 103 as the upper limit and then no need to mess with the GOOS stuff at all, no need to pass it, has half as many tests, etc. Getting the extra 4 chars on linux isn't worth the extra code, especially as linux's "/tmp" already gives a MUCH larger modern to not need truncation in the first place.

Reply:
> New thought. How about just skip this entirely? Just set 103 as the upper limit and then no need to mess with the GOOS stuff at all, no need to pass it, has half as many tests

Agreed.

Reply:
Swapped to 103.

Reply:
> Why change this to an anonymous function? Not objecting, but want to know the reasoning.

I've started to see this pattern popping up and am a fan.

The default and then immediately change it pattern is interesting because in this case, 99% of the time we will be on linux so making that first reads easier.

Either way, removed now.